policy/policy_handler/subscription.go

```diff
@@ -34,19 +34,47 @@ func getSubscriptionData(_ context.Context, req skill.RequestContext) (*goals.Ev
 		SubscriptionBasisT: req.Event.Context.Subscription.Metadata.AfterBasisT,
 	}
 
-	sbom, err := createSbomFromSubscriptionResult(evalMeta.SubscriptionResult, req)
+	sb, err := createSBOMFromManifest(ctx, evalMeta.SubscriptionResult)
+	if err == nil {
+		return evalMeta, req.Event.Context.Subscription.Configuration, sb, nil
+	}
+
+	sb, err = createSBOMFromSubscriptionResult(req, evalMeta.SubscriptionResult)
 	if err != nil {
 		return nil, skill.Configuration{}, nil, fmt.Errorf("failed to create SBOM from subscription result: %w", err)
 	}
 
-	return evalMeta, req.Event.Context.Subscription.Configuration, &sbom, nil
+	return evalMeta, req.Event.Context.Subscription.Configuration, sb, nil
+}
+
+func createSBOMFromManifest(ctx context.Context, subscriptionResult []map[edn.Keyword]edn.RawMessage) (*types.SBOM, error) {
+	imageEdn, ok := subscriptionResult[0][edn.Keyword("image")]
+
+	if !ok {
+		return nil, fmt.Errorf("image not found in subscription result")
+	}
+
+	image := util.Decode[goals.ImageSubscriptionQueryResult](imageEdn)
+
+	ref := image.ImageRepo.Repository
```

Comment:
I think we should check if `ImageRepo` is nil. Subscriptions may not include it.